src/storm/solver/helper/GuessingValueIterationHelper.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <vector>
+
+#include "storm/solver/helper/ValueIterationOperator.h"
+
+#include "storm/solver/OptimizationDirection.h"
+#include "storm/solver/SolverStatus.h"
+#include "storm/storage/BitVector.h"
+
+namespace storm::solver::helper {
+
+template<typename ValueType>
+struct GVIData {
+    std::vector<ValueType> const& x;
+    std::vector<ValueType> const& y;
+    SolverStatus const status;
+};
+
+namespace gviinternal {
+
+typedef uint32_t IndexType;
+
+template<typename ValueType>
+class IterationHelper {
+   public:
+    IterationHelper(storm::storage::SparseMatrix<ValueType> const& matrix);
+
+    IndexType selectRowGroupToGuess(const std::vector<ValueType>& lowerX, const std::vector<ValueType>& upperX);
+
+    ValueType getMaxLength(std::vector<ValueType>& lower, std::vector<ValueType>& upper);
+    ValueType getSumLength(std::vector<ValueType>& lower, std::vector<ValueType>& upper);
+
+   private:
+    std::vector<ValueType> matrixValues;
+    std::vector<IndexType> matrixColumns;
+    std::vector<IndexType> rowIndications;
+    std::vector<uint64_t> const* rowGroupIndices;
+
+    void swipeWeights(std::vector<ValueType>& weights);
+};
+
+}  // namespace gviinternal
+
+template<typename ValueType, OptimizationDirection Dir>
+class GVIBackend;
+
+using IndexType = gviinternal::IndexType;
```

Comment:
I'd be in favour of avoiding this type definition within the storm::solver::helper namespace. Same for `VerifyResult`